### PR TITLE
Prevent early failout if retrieving responder fails

### LINKED ISSUE #44880: ServiceNow plugin does not check all rotation schedules
Expected behavior:

If a schedule has no on-call users it will check all schedules for auto-approving.  Having multiple on-call schedules is typical. A person may be on more then one rotation and it should auto-approve them for any of the rotations they are on.

Current behavior:

If multiple schedules are listed as rotation ids if there are no on-call users it will exit checking and throw an error.

```yaml
      annotations:
        teleport.dev/schedules:
        - 1234
        - 4321
 ```

Log example:
```bash
Jul 30 01:14:03 ftest.teleportdemo.com teleport[552]: 2024-07-30T01:14:03Z DEBU             Checking the following shifts to see if the requester is on-call: [501ae7ada8812300964fb5fe74c31e3c 87503b21a8c12300964fb5fe74c31ec0] request_id:01910133-9356-7e6b-ab68-260c57b06827 request_op:put request_state:PENDING servicenow/app.go:403
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: 2024-07-30T01:14:04Z ERRO             Failed to process request error:[
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: ERROR REPORT:
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: Original Error: *trace.NotFoundError no user found for given rota: &#34;87503b21a8c12300964fb5fe74c31ec0&#34;
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: Stack Trace:
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/access/servicenow/client.go:234 github.com/gravitational/teleport/integrations/access/servicenow.(*Client).GetOnCall
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/access/servicenow/app.go:441 github.com/gravitational/teleport/integrations/access/servicenow.(*App).getOnCallUsers
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/access/servicenow/app.go:405 github.com/gravitational/teleport/integrations/access/servicenow.(*App).tryApproveRequest
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/access/servicenow/app.go:288 github.com/gravitational/teleport/integrations/access/servicenow.(*App).onPendingRequest
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/access/servicenow/app.go:210 github.com/gravitational/teleport/integrations/access/servicenow.(*App).onWatcherEvent
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/lib/watcherjob/watcherjob.go:278 github.com/gravitational/teleport/integrations/lib/watcherjob.job.eventLoop.job.eventFuncHandler.func1
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/lib/process.go:213 github.com/gravitational/teleport/integrations/lib.jobFunc.DoJob
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         github.com/gravitational/teleport/integrations/lib/process.go:101 github.com/gravitational/teleport/integrations/lib.NewProcess.func2.1
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]:         runtime/asm_amd64.s:1695 runtime.goexit
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: User Message: no user found for given rota: &#34;87503b21a8c12300964fb5fe74c31ec0&#34;] request_id:01910133-9356-7e6b-ab68-260c57b06827 request_op:put request_state:PENDING servicenow/app.go:219
Jul 30 01:14:04 ftest.teleportdemo.com teleport[552]: 2024-07-30T01:14:04Z INFO [AUDIT]     access_request.search cluster_name:ftest.teleportdemo.com code:T5004I ei:0 event:access_request.search namespace:default predicate_expression:resource.metadata.name == "ceebb833-d5
```

Bug details:
- Teleport version: 16.1.0
- Recreation steps

  1. Configure a servicenow hosted plugin 
  2. Within servicenow have multiple on-call schedules with different times
  3. Set the rotation ids for at least two on-call schedules
  4. Submit a access request that uses the rotation ids.

### PR DESCRIPTION
**Pull request by EdwardDowling**
Fixes #44880 
Prevents Servicenow plugin from returning early if checking an on-call schedule fails

### CODE REVIEW

**Comment by zmb3** [comment]
Test coverage?

File: integrations/access/servicenow/app.go [hunk 1]
```diff
@@ -436,11 +436,13 @@ func (a *App) tryApproveRequest(ctx context.Context, req types.AccessRequest) er
...
 
 func (a *App) getOnCallUsers(ctx context.Context, serviceNames []string) ([]string, error) {
+	log := logger.Get(ctx)
 	onCallUsers := []string{}
 	for _, scheduleName := range serviceNames {
 		respondersResult, err := a.serviceNow.GetOnCall(ctx, scheduleName)
 		if err != nil {
-			return nil, trace.Wrap(err)
+			log.WithError(err).Error("Failed to retrieve responder from schedule")
+			continue
```
Comment (marcoandredinis) [inline integrations/access/servicenow/app.go:445]: Should we distinguish `not found` errors from other types of errors?
Also, can you please add test coverage for this change?

File: integrations/access/servicenow/client.go [hunk 2]
```diff
@@ -42,6 +42,24 @@ const (
...
+	ResolveIncident(ctx context.Context, incidentID string, resolution Resolution) error
+	// GetOnCall returns the current users on-call for the given rota ID.
+	GetOnCall(ctx context.Context, rotaID string) ([]string, error)
+	// GetUserName returns the name for the given user ID
+	GetUserName(ctx context.Context, userID string) (string, error)
+	// CheckHealth pings servicenow to check if it is reachable.
+	CheckHealth(ctx context.Context) error
+	buildIncidentBody(webProxyURL *url.URL, reqID string, reqData RequestData) (string, error)
+	buildReviewNoteBody(review types.AccessReview) (string, error)
+	buildResolutionNoteBody(resolution Resolution) (string, error)
```
Comment (marcoandredinis) [inline integrations/access/servicenow/client.go:60]: These private methods in an interface doesn't sound right.
From the looks of it, we can easily make those methods not depend on the client, like they do now.
Eg, `buildResolutionNoteBody` needs an extra argument for the `CloseCode`

Comment (EdwardDowling) [reply integrations/access/servicenow/client.go]: Something like this?

File: integrations/access/servicenow/app.go (hunk 1 above)
Comment (EdwardDowling) [reply integrations/access/servicenow/app.go]: Is this along the lines of what you were thinking?

Comment (marcoandredinis) [reply integrations/access/servicenow/app.go]: Yes, thank you.

File: integrations/access/servicenow/client.go (hunk 2 above)
Comment (marcoandredinis) [reply integrations/access/servicenow/client.go]: Yes, thank you

**Review by marcoandredinis — APPROVED** [review]

**Comment by EdwardDowling** [comment]
@tigrato Can you take a look at this when you get a chance?

File: integrations/access/servicenow/app.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ type App struct {
 
 	PluginName string
 	teleport   teleport.Client
-	serviceNow *Client
+	serviceNow ServicenowClient
```
Comment (zmb3) [inline integrations/access/servicenow/app.go:67]:
```suggestion
	serviceNow ServiceNowClient
```

Perhaps?

**Review by zmb3 — APPROVED** [review]

**Pull request merged**